utils/ssh.py

```diff
@@ -213,20 +215,31 @@ def client_address(self):
     def appliance_has_netapp(self):
         return self.run_command("stat /var/www/miq/vmdb/HAS_NETAPP").rc == 0
 
-    def install_ssh_keys(self):
+    def install_ssh_keys(self, padding=0):
+        self.wait_ssh()
         self._keystate = _ssh_keystate.installing
         if not _ssh_key_file.check():
             keygen()
         self._connect_kwargs['key_filename'] = _ssh_key_file.strpath
 
-        if self.run_command('test -f ~/.ssh/authorized_keys').rc != 0:
-            self.run_command('mkdir -p ~/.ssh')
+        if self.run_command('test -f ~/.ssh/authorized_keys', padding=padding + 1).rc != 0:
+            self.run_command('mkdir -p ~/.ssh', padding=padding + 1)
             self.put_file(_ssh_key_file.strpath, '~/.ssh/id_rsa')
             self.put_file(_ssh_pubkey_file.strpath, '~/.ssh/id_rsa.pub')
             self.run_command('cp ~/.ssh/id_rsa.pub ~/.ssh/authorized_keys;'
-                'chmod 700 ~/.ssh; chmod 600 ~/.ssh/*')
+                'chmod 700 ~/.ssh; chmod 600 ~/.ssh/*', padding=padding + 1)
         self._keystate = _ssh_keystate.installed
 
+    def wait_ssh(self):
+        def _wait_f():
+            try:
+                return self.run_command("true", log=False).rc == 0
+            except (EOFError, paramiko.SSHException) as e:
```

Comment:
I've only ever seen an EOFError happen when the appliance is misconfigured, and will (probably?) never recover. SSHExceptions are usually unrecoverable as well. Do you have a repeatable test case where waiting longer like this makes a failing SSH eventually start to work?